redis.conf

```diff
@@ -1826,6 +1826,14 @@ activerehashing yes
 # if the size of the output buffers reach 32 megabytes, but will also get
 # disconnected if the client reaches 16 megabytes and continuously overcomes
 # the limit for 10 seconds.
+# A soft limit is also used to postpone execution of pipelined commands and
+# allow response to be written to the socket (even when soft limit seconds is 0)
+# I.E. a soft limit seconds of 0 means the soft limit will put a back pressure
+# to pipelined commands without disconnecting the a client.
+#
+# A negative limit (soft or hard), means an automatically calculated value which
+# is a percentage of the maxmemory limit.
+# Both the hard or the soft limit can be disabled by setting them to zero.
```

Comment:
We should consider having a percentage based approach for other values as well, I like this approach.

Reply:
like what?
for instance, these are probably not a good idea: `proto-max-bulk-len`, `repl-backlog-size`, `hash-max-ziplist-value`
these, maybe, but i don't see much value: `client-query-buffer-limit`, `active-defrag-ignore-bytes`

Reply:
I guess I thought there was more that made sense when I made this comment. I was thinking about how a customer had asked us to support max client side tracking table as a percentage of memory, but that would still probably require a separate value.